agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/AgentImplementation.java

```diff
@@ -153,6 +153,8 @@ public static String getAgentJarLocation() throws UnsupportedEncodingException {
         }
 
         String path = AgentImplementation.class.getProtectionDomain().getCodeSource().getLocation().getPath();
```

Comment:
Use https://docs.oracle.com/javase/8/docs/api/java/nio/file/Paths.html instead of string?